BLSQ/iaso
# IA-1609 IA-1793: instances dynamic fields & query builder fields

Column selection code was leading to infinite loop, code has to be refactored
Possible fields for dynamic columns and query builder should be available while selecting a form

Related JIRA tickets : IA-1609, IA-1793

## Self proof reading checklist


## Changes

- Big refactor of columns selections on submissions pages (typescript)
- Move Button to select columns out of TopBar
- Split fetch of possible fields for query builder and columns selection drawer

## How to test

Go to submissions page.
Change form selection 
Make sure you have possible fields in the query builder without having to click on search
Columns selection button should be disabled
Click on search
Columns selection button should be enabled
Click on it
Same possible fields should be visible in the right drawer.
Play a bit with the tool

## Print screen / video

<img width="746" alt="Screenshot 2023-01-25 at 13 59 45" src="https://user-images.githubusercontent.com/12494624/214611123-ddf18d1e-8cd9-41bb-a926-8e69c1b1cdac.png">
